packages/capabilities/src/http.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * HTTP Capabilities.
+ *
+ * These can be imported directly with:
+ * ```js
+ * import * as HTTP from '@web3-storage/capabilities/http'
+ * ```
+ *
+ * @module
+ */
+import { capability, Schema, ok } from '@ucanto/validator'
+import { blobStruct } from './blob.js'
+import { equal, equalBlob, equalWith, SpaceDID, and } from './utils.js'
+
+/**
+ * `http/put` capability invocation MAY be performed by any agent on behalf of the subject.
+ * The `blob/add` provider MUST add `/http/put` effect and capture private key of the
+ * `subject` in the `meta` field so that any agent could perform it.
+ */
+export const put = capability({
+  can: 'http/put',
+  /**
+   * DID of the (memory) space where Blob is intended to
+   * be stored.
+   */
+  with: SpaceDID,
+  nb: Schema.struct({
+    /**
+     * Blob to allocate on the space.
+     */
+    blob: blobStruct,
```

Comment:
@Gozala note that I may change this to get the size info from `content length` as you suggested

Reply:
```suggestion
    content,
```